adding an iOS build to travis

csomor opened the pull request.

csomor committed:
fixing compilation of tests that cannot build

bracket code with the corresponding wxUSE… macros

csomor committed:
adding directive for iOS

csomor committed:
adding a switch for skipping the run - not the build of tests

right now I don’t know yet, how to run the test binary in the iOS simulator, but building it is still a good test in itself

csomor committed:
adding skipping of tests

csomor committed:
increasing minimum deployment to get proper c++17 support

csomor commented:
Hi @vadz, could you please look at the changes, especially the addition to the travis script, whether this is fine with you. Thanks

vadz commented on .travis.yml:
```diff
@@ -7,76 +7,10 @@ sudo: required
         - os: osx
           osx_image: xcode11.4
-          env: wxCONFIGURE_FLAGS="--enable-monolithic --with-cxx=17 --host=i686-apple-darwin_sim --build=x86_64-apple-darwin17.7.0 --with-osx_iphone --with-macosx-version-min=9.0 --with-macosx-sdk=$(xcrun --sdk iphonesimulator --show-sdk-path) --enable-stl --with-libtiff=builtin" wxSKIP_GUI=1 wxSKIP_SAMPLES=1
+          env: wxCONFIGURE_FLAGS="--enable-monolithic --with-cxx=17 --host=i686-apple-darwin_sim --build=x86_64-apple-darwin17.7.0 --with-osx_iphone --with-macosx-version-min=10.0 --with-macosx-sdk=$(xcrun --sdk iphonesimulator --show-sdk-path) --enable-stl --with-libtiff=builtin" wxSKIP_GUI=1 wxSKIP_TESTING=1 wxSKIP_SAMPLES=1
```
Curious, why do we need `--with-libtiff=builtin`? I.e. why do we need it for this one and why don't we need it for the other 3rd party libs?

csomor replied:
I have no idea, but the build broke, it somehow found a system installed tiff include dir, but then failed include tiff.h, the other picture libs were built-in, and also when building on my system configure configure uses the built-in automatically

csomor replied:
configure apparently thinks `-I/usr/local/Cellar/libtiff/4.1.0/include` is perfect, but then it fails with` ./src/common/imagtiff.cpp:40:14: fatal error: 'tiff.h' file not found`

[build log](https://travis-ci.org/github/wxWidgets/wxWidgets/jobs/684169612)

vadz replied:
I see, thanks. I need to look into the proper use of `pkg-config` when cross-compiling, but for now using this option or, better, `--disable-sys-libs` (because system libz seems to be also found), is indeed the right thing to do.

csomor replied:
the libtiff is the only header that is picked up from the outside, I have no idea why. Actually zlib is correctly taken from `-isysroot`

`/Applications/Xcode-11.4.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk/usr/include/zlib.h`

but then again, there is a problem - and we get a proper warning - because according to our deployment settings, inflateValidate may not be available, since the guards are evaluated using macros from the zlib from the SDK, not from the lowest version. So I'll switch that to `-disable-sys-libs`, thanks

vadz commented on build/tools/travis-ci.sh:
```diff
@@ -73,6 +73,11 @@ case $wxTOOLSET in
 
+        if [ "$wxSKIP_TESTING" = 1 ]; then
+            echo 'Skipping running tests'
+            exit 0
```
If possible, I'd like to try at least building the minimal sample, as otherwise we wouldn't detect any breakage resulting in linking errors.

csomor replied:
I'm only skipping **running** the test, test and test_gui are still built, would this not bet catching the linking errors as well ?

vadz replied:
Oops, sorry, I've just misread this in my haste. This is indeed fine, sorry again for the false alarm.

vadz reviewed (commented):
Thanks a lot!

Before this is merged, the rest of the builds in `.travis.yml` need to be restored, but I can do this if you prefer.

Other than that, I'm fine with not running the tests under iOS if this is too broken, but I still hope to do it at some later time, so I'd just like to understand what exactly is the problem with them? I.e. what are the currently failing tests? I could take a look at them if there are not too many of them.

Thanks again!

csomor commented:
> Before this is merged, the rest of the builds in `.travis.yml` need to be restored, but I can do this if you prefer.

I think I'll just copy the few lines and then discard the changes of the file and copy that paragraph back thanks

csomor commented:
> Other than that, I'm fine with not running the tests under iOS if this is too broken

it's not that they are broken, but that they have to be run under the simulator. I have not yet fully grasped how to properly do this entirely from the command line, it may well be that I'll expand my test PR to run iOS tests under the simulator, it might be simpler in the end ...

vadz commented:
To repeat, I'm perfectly fine with merging this already (just after restoring all the other Travis builds) and working on actually running the tests later, please let me know what would you prefer.

Thanks again!

csomor committed:
using --disable-sys-libs, restoring other targets

even when the zlib in -isysroot is used, due to deployment on lower iOS versions inflateValidate may not be available and crash. The guards are evaluated using macros from the zlib from the SDK, not from the lowest version supported.

csomor commented:
as the error in the last travis build is due to me cancelling all other builds, I'm merging ..
